src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayOfStates = [];
+  const copyOfState = { ...state };
+
+  const finiteStateMachine = ({ type, extraData, keysToRemove }) => {
```

Comment:
why did you define a function inside a function, are you going to reuse it?

if you want to leave this solution better to move `finiteStateMachine` to the upper scope